src/include/catalog/schema.h

```diff
@@ -171,6 +173,28 @@ class Schema : public Printable {
     return true;
   }
 
+// For single column default
+  inline bool AllowDefault(const oid_t column_id) const {
+    for (auto constraint : columns[column_id].GetConstraints()) {
+      if (constraint.GetType() == ConstraintType::DEFAULT) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  // Get the default value for the column
+  inline type::Value* GetDefaultValue(const oid_t column_id) const {
+    for (auto constraint : columns[column_id].GetConstraints()) {
+      if (constraint.GetType() == ConstraintType::DEFAULT) {
+        return constraint.getDefaultValue();
+      }
+    }
+
+    return nullptr;
+  }
+
   // Add constraint for column by id
   inline void AddConstraint(oid_t column_id,
```

Comment:
Two AddConstraint functions, one for a given column id, and another to deal with column names.